src/libraries/Microsoft.Extensions.Configuration.Binder/tests/SourceGenerationTests/Baselines/TestConfigureCallGen.generated.txt

```diff
@@ -1,91 +1,124 @@
 // <auto-generated/>
 #nullable enable
 
-using System.Linq;
-
 internal static class GeneratedConfigurationBinder
 {
-    public static global::Microsoft.Extensions.DependencyInjection.IServiceCollection Configure<T>(this global::Microsoft.Extensions.DependencyInjection.IServiceCollection services, global::Microsoft.Extensions.Configuration.IConfiguration configuration)
+    public static global::Microsoft.Extensions.DependencyInjection.IServiceCollection Configure<T>(this global::Microsoft.Extensions.DependencyInjection.IServiceCollection services, global::Microsoft.Extens
+ions.Configuration.IConfiguration configuration)
     {
-        if (typeof(T) == typeof(Program.MyClass))
+        if (configuration is null)
+        {
+            throw new global::System.ArgumentNullException(nameof(configuration));
+        }
+
+        if (typeof(T) == typeof(global::Program.MyClass))
         {
-            return services.Configure<Program.MyClass>(obj =>
+            return services.Configure<global::Program.MyClass>(obj =>
             {
+                if (!global::Microsoft.Extensions.Configuration.Binder.SourceGeneration.Helpers.HasValue(configuration))
+                {
+                    return default;
```

Comment:
This is an `Action`, which means it returns `void`. Shouldn't this just be `return;`?

Reply:
Right. I didn't add e2e tests for `Configure` in the initial PR because it's implemented in `Microsoft.Extensions.Options.ConfigurationExtensions`. But we should add tests to both assemblies - https://github.com/dotnet/runtime/issues/83621.